devicetypes/smartthings/zwave-metering-dimmer.src/zwave-metering-dimmer.groovy

```diff
@@ -16,210 +16,309 @@
+        zwave.basicV1.basicSet(value: 0x00),
+        zwave.switchMultilevelV1.switchMultilevelGet(),
+    ], 5000)
 }
 
 def poll() {
-	delayBetween([
-		zwave.meterV2.meterGet(scale: 0).format(),
-		zwave.meterV2.meterGet(scale: 2).format(),
-	], 1000)
+    encapSequence([
+        meterGet(scale: 0),
+        meterGet(scale: 2),
+    ], 1000)
 }
 
 /**
  * PING is used by Device-Watch in attempt to reach the Device
  * */
 def ping() {
-	log.debug "ping() called"
-	refresh()
+    log.debug "ping() called"
+    refresh()
 }
 
 def refresh() {
-	delayBetween([
-		zwave.switchMultilevelV1.switchMultilevelGet().format(),
-		zwave.meterV2.meterGet(scale: 0).format(),
-		zwave.meterV2.meterGet(scale: 2).format(),
-	], 1000)
+    log.debug "refresh()"
+
+    encapSequence([
+        zwave.switchMultilevelV1.switchMultilevelGet(),
+        meterGet(scale: 0),
+        meterGet(scale: 2),
+    ], 1000)
 }
 
 def setLevel(level) {
-	if(level > 99) level = 99
-	delayBetween([
-		zwave.basicV1.basicSet(value: level).format(),
-		zwave.switchMultilevelV1.switchMultilevelGet().format()
-	], 5000)
+    if(level > 99) level = 99
+    encapSequence([
+        zwave.basicV1.basicSet(value: level),
+        zwave.switchMultilevelV1.switchMultilevelGet()
+    ], 5000)
+}
+
+def configure() {
+    log.debug "configure()"
+    def result = []
+
+    log.debug "Configure zwaveInfo: "+zwaveInfo
+
+    if (zwaveInfo.mfr == "0086") {  // Aeon Labs meter
+        result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 80, size: 1, scaledConfigurationValue: 2)))    // basic report cc
+        result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 101, size: 4, scaledConfigurationValue: 4)))   // report power in watts
+        result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 111, size: 4, scaledConfigurationValue: 60)))  // every 5 min
+        result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 102, size: 4, scaledConfigurationValue: 8)))   // report energy in kWh
+        result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 112, size: 4, scaledConfigurationValue: 60)))  // every 5 min
+        result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 103, size: 4, scaledConfigurationValue: 0)))   // no third report
+        result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 113, size: 4, scaledConfigurationValue: 300))) // every 5 min
+    }
+    result << response(encap(meterGet(scale: 0)))
+    result << response(encap(meterGet(scale: 2)))
+}
+
+def reset() {
+    encapSequence([
+        meterReset(),
+        meterGet(scale: 0)
+    ])
+}
+
+def meterGet(scale)
+{
+    zwave.meterV2.meterGet(scale)
+}
+
+def meterReset()
+{
+    zwave.meterV2.meterReset()
+}
+
+def normalizeLevel(level)
+{
+    // Normalize level between 1 and 100.
+    (int)((float)level * 1.01010101)
```

Comment:
I agree with @ALamchaS that this isn't as clear as explicitly converting 99 to 100

Reply:
Did you want changes to this?

Reply:
@sagar-trility Our general strategy has just been to take level reports of 99 and set them to 100, but to otherwise just report the value as a percent.

Reply:
@sagar-trility ping